pygen/pygen_src/riscv_instr_gen_config.py

```diff
@@ -1,34 +1,175 @@
+        self.max_stack_len_per_program = 16 * (riscv_core_setting.XLEN/8)
+        self.require_signature_addr = 0
+        self.signature_addr = 0xdeadbeef        
+    
+        # lists for mode_exception & mode_interrupt
+        self.mode_exp_lst = list(map(lambda a: a.name, exception_cause_t))
+        self.mode_intrpt_lst = list(map(lambda b: b.name, interrupt_cause_t))
+    
+        # list for init_privileged_mode
+        self.init_privileged_mode = list(map(lambda mode: mode.name, privileged_mode_t))
+    
+        # list of main implemented CSRs
+        self.invalid_priv_mode_csrs = [] 
+     
+        # initialsation for mode dicts
+        for i in self.mode_exp_lst:
+          self.m_mode_exception_delegation[i] = None
+          self.s_mode_exception_delegation[i] = None
+    
+        for j in self.mode_intrpt_lst:
+          self.m_mode_interrupt_delegation[j] = None
+          self.s_mode_interrupt_delegation[j] = None
+  
+  
+    def check_setting(self):
+        support_64b = 0
+        support_128b = 0
+
+        # list of satp_mode_t from riscv_core_setting.py
+        stp_md_lst = riscv_core_setting.SATP_MODE
+
+        # list of riscv_instr_group_t with names of riscv_instr_name_t in it. 
+        supported_isa_lst = list(map(lambda z: z.name, riscv_instr_group_t))
+
+        # list of supported_isa from riscv_core_setting.py
+        supported_isa_stng_lst = riscv_core_setting.supported_isa
+
+        # check the valid isa support 
+        for x in supported_isa_stng_lst: 
+          if x == (supported_isa_lst[1] or supported_isa_lst[3] or supported_isa_lst[5] or \
+                   supported_isa_lst[8] or supported_isa_lst[11] or supported_isa_lst[13] or \
+                   supported_isa_lst[19]):
+            support_64b = 1
+            print(x)
+          elif x == (supported_isa_lst[14] or supported_isa_lst[15]):
+            support_128b = 1
+            print(x)
+             
+        if (support_128b == 1) and (riscv_core_setting.XLEN != 128):
+          logging.critical("XLEN should be set to 128 based on riscv_core_setting.supported_isa setting")
+          print("XLEN Vlaue =", riscv_core_setting.XLEN)
+            
+        if (support_128b == 0) and (support_64b == 1) and (riscv_core_setting.XLEN != 64):
+          logging.critical("XLEN should be set to 64 based on riscv_core_setting.supported_isa setting")
+          print("XLEN Vlaue =",riscv_core_setting.XLEN)
+            
+        if not(support_128b or support_64b) and (riscv_core_setting.XLEN != 32):
+          logging.critical("XLEN should be set to 32 based on riscv_core_setting.supported_isa setting")
+          print("XLEN Vlaue =", riscv_core_setting.XLEN)
+            
+        if not(support_128b or support_64b) and not(('SV32' in stp_md_lst) or ('BARE' in stp_md_lst)):
+          logging.critical("SATP mode %s is not supported for RV32G ISA", stp_md_lst)  
+         
```

Comment:
Use 4 space indentation

nit: avoid print(), use logging.info() for printing

Reply:
done the changes, cleared the spacing errors